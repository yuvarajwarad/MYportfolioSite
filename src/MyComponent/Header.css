/* Header.css */
.monkey {
  position: absolute;
  transition: transform 0.5s ease-in-out, opacity 0.3s ease-in-out;
}

.monkey img {
  width: 60px;
  height: auto;
  cursor: pointer;
}

.monkey.jump {
  animation: jumpAnimation 1s ease-in-out infinite alternate;
  opacity: 1;
}


@keyframes jumpAnimation {
  0% {
    transform: translateY(0);
  }
  100% {
    transform: translateY(-20px);
  }
}
