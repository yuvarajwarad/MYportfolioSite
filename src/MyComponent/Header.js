import React, { useState } from "react";
import "./Header.css"; // Import your CSS file
import { Link } from "react-router-dom";
import monkeyImage from "./monkey.png"; // Import your monkey image
import "../App.css";

const Header = () => {
  const [jumping, setJumping] = useState(false);
  const [position, setPosition] = useState({ top: 0, left: 0 });

  const handleJump = (event) => {
    const link = event.target.getBoundingClientRect();
    setPosition({
      top: link.top + window.scrollY - 60, // Adjust the position
      left: link.left + window.scrollX,
    });
    setJumping(true);

    setTimeout(() => {
      setJumping(false);
    }, 2000);
  };

  return (
    <div className="App">
      <header>
        <nav>
          <div className="yuvi-warad">
            <Link to="/">Yuvraj Warad</Link>
          </div>
          <div className="right">
            <ul>
              <li>
                <Link to="/" onClick={handleJump}>
                  Home
                </Link>
              </li>
              <li>
                <a href="#exp" onClick={handleJump}>
                  Work Experience
                </a>
              </li>
              <li>
                <Link to="/Skills" onClick={handleJump}>
                  My Skills
                </Link>
              </li>
              <li>
                <Link to="/Projects" onClick={handleJump}>
                  Projects
                </Link>
              </li>
              <li>
                <Link to="/Education" onClick={handleJump}>
                  Education
                </Link>
              </li>
              <li>
                <Link to="/About" onClick={handleJump}>
                  About
                </Link>
              </li>

              <li>
                <Link to="/Contact" onClick={handleJump}>
                  Contact ME
                </Link>
              </li>
            </ul>
          </div>
        </nav>
        {jumping && (
          <div
            className="monkey"
            style={{
              top: position.top + "px",
              left: position.left + "px",
            }}
          >
            <img src={monkeyImage} alt="Jumping Monkey" />
          </div>
        )}
      </header>
    </div>
  );
};

export default Header;
