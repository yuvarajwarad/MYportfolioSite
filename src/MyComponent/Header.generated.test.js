import React from "react";
import { render, screen } from "@testing-library/react"; // Import render and screen from @testing-library/react
import Header from "./Header";

jest.mock("./Header.css");
jest.mock("react-router-dom");
jest.mock("./monkey.png");
jest.mock("../App.css");

describe("<Header>", () => {
  it("should render component", () => {
    render(<Header />); // Render the Header component

    // Use screen.getByText to find the "Work Experience" text and assert its presence
    const linkElement = screen.getByText(/Work Exper/i);
    expect(linkElement).toBeInTheDocument();
  });
});
