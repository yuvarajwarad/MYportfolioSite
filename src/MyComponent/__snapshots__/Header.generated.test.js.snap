// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> should render component 1`] = `
<div
  className="App"
>
  <header>
    <nav>
      <div
        className="yuvi-warad"
      />
      <div
        className="right"
      >
        <ul>
          <li />
          <li>
            <a
              href="#exp"
              onClick={[Function]}
            >
              Work Experience
            </a>
          </li>
          <li />
          <li />
          <li />
          <li />
          <li />
        </ul>
      </div>
    </nav>
  </header>
</div>
`;
